src/components/CartsContainer/CartsContainer.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { confusedIMG } from '../../assets/resources/imagesURL';
+import { deleteCurrentPhoneCart, clearCurrentCart } from '../../store/actions/inCartAction';
+import './CartsContainer.scss';
+
+const CartsContainer = (props) => {
+
+   const { cart, totalAmount, deleteCurrentPhoneCart, clearCurrentCart } = props;
+   const testIMG = 'https://s.abcnews.com/images/Technology/samsung-phone-gty-jpo-190423_hpMain_16x9_992.jpg';
+
+   const clearCartHandler = () => {
+      clearCurrentCart([]);
+   }
+
+   const deleteCartHandler = (product, indexItem) => {
+      let filtered = cart.filter((item, index) => index !== indexItem);
+      deleteCurrentPhoneCart(filtered);
+   }
```

Comment:
State change should be handled inside of reducer

pass only the `indexItem` to `deleteCurrentPhoneCart` and filter the cart inside of reducer